km/km_decode.c

```diff
@@ -834,6 +860,488 @@ static void decode_opcode(km_vcpu_t* vcpu, x86_instruction_t* ins)
+                                  0x19,   // VEXTRACTF128 (pp = 1)
+                                  0x1d,   // VCVTPS2PH (pp = 1)
+                                  0x20,   // VPINSRB (pp = 1)
+                                  0x21,   // VINSERTPS (pp = 1)
+                                  0x22,   // VPINSRD, VPINSRQ (pp = 1)
+                                  0x38,   // VINSERTI128 (pp = 1)
+                                  0x39,   // VEXTRACTI128 (pp = 1)
+                                  0x40,   // VDPPS (pp = 1)
+                                  0x41,   // VDPPD (pp = 1)
+                                  0x42,   // VMPSADBW (pp = 1)
+                                  0x44,   // VPCLMULQDQ (pp = 1)
+                                  0x46,   // VPERM2I128 (pp = 1)
+                                  0x48,   // VPERMILzz2PS (pp = 1)
+                                  0x49,   // VPERMILzz2PD (pp = 1)
+                                  0x4a,   // VBLENDVPS (pp = 1)
+                                  0x4b,   // VBLENDVPD (pp = 1)
+                                  0x4c,   // VPBLENDVB (pp = 1)
+                                  0x5c,   // VFMADDSUBPS (pp = 1)
+                                  0x5d,   // VFMADDSUBPD (pp = 1)
+                                  0x5e,   // VFMSUBADDPS (pp = 1)
+                                  0x5f,   // VFMSUBADDPD (pp = 1)
+                                  0x60,   // VPCMPESTRM (pp = 1)
+                                  0x61,   // VPCMPESTRI (pp = 1)
+                                  0x62,   // VPCMPISTRM (pp = 1)
+                                  0x63,   // VPCMPISTRI (pp = 1)
+                                  0x78,   // VFNMADDPS (pp = 1)
+                                  0x79,   // VFNMADDPD (pp = 1)
+                                  0x7a,   // VFNMADDSS (pp = 1)
+                                  0x7b,   // VFNMADDSD (pp = 1)
+                                  0x7c,   // VFNMSUBPS (pp = 1)
+                                  0x7d,   // VFNMSUBPD (pp = 1)
+                                  0x7e,   // VFNMSUBSS (pp = 1)
+                                  0x7f,   // VFNMSUBSD (pp = 1)
+                                  0xdf,   // VAESKEYGENASSIST (pp = 1)
+                                  0xf0,   // RORX (pp = 3)
+                                  OPCODE_MAP_END};
+
+static const ushort* vex_maps[32] = {   // 1-3 are valid indexes
+    NULL,
+    vex_map1,
+    vex_map2,
+    vex_map3};
+
+// We don't support the xop maps yet.
+// But the code is plumbed to use them since they are about the same as
+// VEX_3BYTE_PREFIX instruction prefixes.
+// In the amd manual, tables A-25 through A-31 specify the xop opcode maps
+// and the xop opcode groups
+const unsigned char* xop_groups[32];
+const ushort* xop_maps[32];   // 8-10 are valid indexes
+
+static unsigned short find_opcode_in_map(const ushort* map, unsigned char opcode)
+{
+   const ushort* p;
+
+   for (p = map; *p != OPCODE_MAP_END; p++) {
```

Comment:
are you sure you want to search a table on each fault?

Reply:
When exception processing becomes a normal programming technique then we will need to change.
Given how people seem to think it wouldn't surprise me if that is what the industry decided to do.